### PR TITLE
Use cargo workspace features

### PR DESCRIPTION
**Pull request by bilelmoussaoui**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -23,3 +23,27 @@ members = [
...
+]
+edition = "2021"
+rust-version = "1.70"
+
+[workspace.dependencies]
+once_cell = "1.19"
+libc = "0.2"
+bitflags = "2.4"
+thiserror = "1"
+gir-format-check = "^0.1"
```
Comment (sdroege) [inline Cargo.toml:44]: This should be in build dependencies probably?

File: gdk-pixbuf/sys/Cargo.toml [hunk 2]
```diff
@@ -28,17 +28,34 @@ v2_42 = ["v2_40"]
...
 keywords = ["gdk-pixbuf", "ffi", "gtk-rs", "gnome"]
-license = "MIT"
 name = "gdk-pixbuf-sys"
-repository = "https://github.com/gtk-rs/gtk-rs-core"
-version = "0.19.0"
-edition = "2021"
-rust-version = "1.70"
+
+[package.authors]
+workspace = true
```
Comment (sdroege) [inline gdk-pixbuf/sys/Cargo.toml:37]: Is this left alone by `gir`?

File: Cargo.toml (hunk 1 above)
Comment (bilelmoussaoui) [reply Cargo.toml:44]: You can't define build dependencies inside a workspace's Cargo.toml; crates have to use `gir-formart-check.worskspace = true`

Comment (sdroege) [reply Cargo.toml:44]: dev-dependencies

File: gdk-pixbuf/sys/Cargo.toml (hunk 2 above)
Comment (bilelmoussaoui) [reply gdk-pixbuf/sys/Cargo.toml:37]: yes, so annoying

**Review by sdroege — APPROVED** [review]
Nice :)

File: Cargo.toml (hunk 1 above)
Comment (bilelmoussaoui) [reply Cargo.toml:44]: such key doesn't exist :/ you just define everything the main Cargo.toml and opt-in in sub-crates

Comment (sdroege) [reply Cargo.toml:44]: Ok then :) Nothing we can do!

**Review by sdroege — APPROVED** [review]

**Pull request merged**